README.md

```diff
@@ -36,6 +36,14 @@ var json = await response.Content.ReadAsStringAsync();
 Console.Write(json); // {'name' : 'Test McGee'}
 ```
 
+You can also use the dotnet DI system to inject the MockHttpMessageHandler:
+```csharp
+services
+    .AddSingleton<MockHttpMessageHandler>()
+    .AddHttpClient<YourApiClient>()
+        .ConfigurePrimaryHttpMessageHandler<MockHttpMessageHandler>();
+```
```

Comment:
Would it make more sense to show an MockHttpMessageHandler be instatiated and resolved from a factory in ConfigurePrimaryHttpMessageHandler?

Otherwise there would be no way to access the handler to setup expectations etc.

Reply:
The test can resolve the MockHttpMessageHandler from the DI container and set expectations. I'm using the Xunit.DependencyInjection like this:
```csharp
public class MyTests
{
    private MockHttpMessageHandler _httpMessageHandlerMock;
    
    public MyTests(MockHttpMessageHandler httpMessageHandlerMock)
    {
         _httpMessageHandlerMock = httpMessageHandlerMock;
    }

    [Fact]
    public void MyTest()
    {
        _httpMessageHandlerMock.ResetExpectations();
        _httpMessageHandlerMock.Expect(...
       // Act
       // Assert
    }
}
```

Setting the MockHttpMessageHandler as a Singleton and calling ResetExpectations() at the beginning of each test is important because the HttpClientFactory caches the MessageHandlers between scopes.

Reply:
Ah okay, it could be worth giving some of that wider context in the example then, as that isn't built-in functionality, and while it may be a common scenario, it could be confusing/misleading for those unfamiliar with it.

Reply:
I made an update with some more content. Is it satisfactory?

Reply:
That looks good to me. I'm not a maintainer on this repo, so I can't approve only @richardszalay can.